File: tmp/src/main/java/com/hades/example/android/MainActivity.java
package com.hades.example.android;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

public class MainActivity extends AppCompatActivity {

    View mMediaBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EventBus.getDefault().register(this);

        setContentView(R.layout.activity_main);

        findViewById(R.id.open_one_video).setOnClickListener(v -> openOneVideo());
        findViewById(R.id.open_one_audio).setOnClickListener(v -> openOneAudio());
        findViewById(R.id.expand).setOnClickListener(v -> clickExpand());
        findViewById(R.id.close).setOnClickListener(v -> clickClose());

        mMediaBar = findViewById(R.id.media_bar);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        EventBus.getDefault().unregister(this);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onCollapseMediaEvent(CollapseVideoEvent event) {
//        EventBus.getDefault().cancelEventDelivery(event);
        showMediaBar();
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onCloseMediaEvent(CloseVideoEvent event) {
//        EventBus.getDefault().cancelEventDelivery(event);
        hideMediaBar();
    }

    private void showMediaBar() {
        mMediaBar.setVisibility(View.VISIBLE);

        mMediaBar.clearAnimation();
        Animation animation = AnimationUtils.loadAnimation(this, R.anim.media_bar_enter_from_bottom_to_top);
        mMediaBar.startAnimation(animation);
    }

    private void hideMediaBar() {
        mMediaBar.setVisibility(View.GONE);

        mMediaBar.clearAnimation();
        Animation animation = AnimationUtils.loadAnimation(this, R.anim.media_bar_exit_from_top_to_bottom);
        mMediaBar.startAnimation(animation);
    }

    private void openOneVideo() {
        startActivity(new Intent(this, SecondActivity.class));
    }

    private void openOneAudio() {
        showMediaBar();
    }

    private void clickExpand() {
        startActivity(new Intent(this, SecondActivity.class));
    }

    private void clickClose() {
        hideMediaBar();
    }
}

File: lib-examples/data-binding-example/src/main/java/com/hades/example/android/libexamples/data_binding/MainActivity.kt
package com.hades.example.android.libexamples.data_binding

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.hades.example.android.libexamples.data_binding.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    //    var user = User("", "");
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // way 1, START
        // binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        // way 1, END
        // way 2, START
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        // way 2, END

        // Binding data
        binding.user = User("Test", "User")
//        binding.user = user
//        user.firstName = "Test"
//        user.lastName = "User"

        findViewById<View>(R.id.setData).setOnClickListener { setData() }
        findViewById<View>(R.id.clearData).setOnClickListener { clearData() }
    }

    private fun setData() {
        binding.user = User("Test first name", "User")
//        user.firstName = "Test first name"
    }

    private fun clearData() {
        binding.user = User("", "")
//        user.firstName = ""
    }
}